### PR TITLE
Add search filter functionality

### PR DESCRIPTION
**Pull request by Craphtex**
This feature allows for filtering the invoices.

The implementation allows for filtering by many fields at the same time.

See meeting notes from 2016-04-19.

### CODE REVIEW

**Comment by Craphtex** [comment]
The functionality is there, even if the surrounding code would need some refactoring.
Will work on that in a chore instead.

File: client/invoice_table.js [hunk 1]
```diff
@@ -1,51 +1,84 @@
...
   helpers: {
     invoices() {
-      return InvoiceTicketsCollection.find(getFilter(date.get(), days.get()), {
+      let filter = MongoDBQueryFactory.mergeDBFilters(MongoDBQueryFactory.createTimeFilter(date.get(), days.get()),MongoDBQueryFactory.createSearchFilter(Template.instance().state.searchTerms()));
+      return InvoiceTicketsCollection.find(filter, {
         sort: {
-          createdAt: sortState.get('sortCreatedAt', true),
-          total: sortState.get('sortTotal', true)
+          createdAt: Template.instance().sortingToDBSorting[StateMachine.get('sortCreatedAt', 'Asc')],
+          total: Template.instance().sortingToDBSorting[StateMachine.get('sortTotal', 'Asc')]
```
Comment (DominikGuzei) [inline client/invoice_table.js:13]: Is there a reason why you are not using `this` to reference the `Template.instance()`?

**Comment by DominikGuzei** [comment]
Hey Christian!
There are some good ideas but also some over-engineering / different implementation for the task that was asked for (there is only a single search box with a category dropdown in the wireframes) 😉 

I like your query building code though! it can be refactored to a better api but the general idea is solid and i think we will use this as a good starting point!

**Comment by DominikGuzei** [comment]
Btw. a good pull request includes a proper description and in this case i would have liked some screenshots too. Remember, the key is communication – it's often better to be over-explicit with everything when working remotely.

File: client/invoice_table.js (hunk 1 above)
Comment (Craphtex) [inline client/invoice_table.js:13]: Think it's a leftover from previous debugging.

**Pull request merged**